data/campaigns/emp04.wmf/scripting/mission_thread.lua

```diff
@@ -72,6 +72,8 @@ end
 -- the chaotic road network has to be cleared
 function clear_roads()
    local o = add_campaign_objective(obj_clear_roads)
+   local o1 = add_campaign_objective(obj_clear_roads_2)
+   o1.done = true
```

Comment:
This design looks a bit weird to me. IMHO it would be cleaner to not define `o1` at this point yet (`local o1 = nil`), then change line 107 use `o1 = add_campaign_objective(...)` and line 113 to `if o1 then ... done`.

Or even better: According to the doc, `LuaObjective::body` is RW so instead of having two separate objectives you could simply change the first objective's text if needed.

Reply:
Well, changing the text of an objective in mission_thread.lua that is defined in texts.lua would be not very understandable imho. 
And I made good experiences with switchinmg objectives visibility based on external events.
implementing your first suggestion would make no difference though as we would have the 2 objectives and need to control their visibility.

Reply:
The way you're doing it here still triggers an additional autosave though doesn't it? The filename of this autosave will suggest that the Clear Roads objective has already been fulfilled which is misleading.

For the clarity you could define the new body as a global variable in texts.lua (just like the objectives) and simply do `o.body = obj_clear_roads_body_2` in the mission_thread. Or stick this in a function `elaborate_clear_roads_objective_text` which is defined in texts.lua.

Reply:
ok, 
I forgot about the autosaves. so I will change this as suggested

Reply:
fixed a second instance of this superfluous savegame as well.